presto-orc/src/main/java/com/facebook/presto/orc/OrcReaderOptions.java

```diff
@@ -13,38 +13,35 @@
  */
 package com.facebook.presto.orc;
 
+import com.google.common.collect.ImmutableMap;
 import io.airlift.units.DataSize;
 
+import java.util.Map;
+
 import static java.util.Objects.requireNonNull;
 
 public class OrcReaderOptions
 {
     private final DataSize maxMergeDistance;
     private final DataSize tinyStripeThreshold;
     private final DataSize maxBlockSize;
-    private final boolean zstdJniDecompressionEnabled;
-    private final boolean mapNullKeysEnabled;
-    private final boolean enableTimestampMicroPrecision;
+    private final Map<OrcReaderFeature, Boolean> featureFlags;
```

Comment:
A set is good enough?

Reply:
Enabled if in the key is in the set, and disable otherwise? I don't see a big difference, but I can change it to a set.